compiler.cpp

```diff
@@ -402,14 +402,13 @@ struct Compilation {
     ~Compilation() = default;
 
     double Cost(const CostPair& pair, const Node& node) {
-        return node->ScriptSize() + Mul(p, pair.sat) + Mul(q, pair.nsat);
+        return node->GetScriptSize() + Mul(p, pair.sat) + Mul(q, pair.nsat);
     }
 
     void Add(const CostPair& pair, Node node) {
         auto new_typ = node->GetType();
         double cost = Cost(pair, node);
-        if (!node->CheckOpsLimit()) return;
-        if (node->GetStackSize() > MAX_STANDARD_P2WSH_STACK_ITEMS) return;
+        if (!node->CheckOpsLimit() || !node->CheckScriptSize() || !node->CheckStackSize()) return;
```

Comment:
I was going to comment that you should add `CheckTimeLocksMix()` here too, but actually let's just replace everything by `IsSafeTopLevel()` to avoid the redundancy?

Reply:
Done